awscli/customizations/dynamodb/types.py

```diff
@@ -0,0 +1,290 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from collections import Mapping, Set
+from decimal import Decimal, Context, Clamped
+from decimal import Overflow, Inexact, Underflow, Rounded
+
+from botocore.compat import six
+
+
+STRING = 'S'
+NUMBER = 'N'
+BINARY = 'B'
+STRING_SET = 'SS'
+NUMBER_SET = 'NS'
+BINARY_SET = 'BS'
+NULL = 'NULL'
+BOOLEAN = 'BOOL'
+MAP = 'M'
+LIST = 'L'
+
+
+DYNAMODB_CONTEXT = Context(
+    Emin=-128, Emax=126, prec=38,
+    traps=[Clamped, Overflow, Inexact, Rounded, Underflow]
+)
+
+
+BINARY_TYPES = (bytearray, six.binary_type)
+
+
+class Binary(object):
```

Comment:
Through the console, I added a binary type and got the following traceback back for it:
```
Traceback (most recent call last):
  File "/Users/kyleknap/GitHub/aws-cli/awscli/clidriver.py", line 260, in main
    return command_table[parsed_args.command](remaining, parsed_args)
  File "/Users/kyleknap/GitHub/aws-cli/awscli/customizations/commands.py", line 190, in __call__
    parsed_globals)
  File "/Users/kyleknap/GitHub/aws-cli/awscli/customizations/commands.py", line 187, in __call__
    return self._run_main(parsed_args, parsed_globals)
  File "/Users/kyleknap/GitHub/aws-cli/awscli/customizations/dynamodb/subcommands.py", line 118, in _run_main
    self._select(parsed_args, parsed_globals)
  File "/Users/kyleknap/GitHub/aws-cli/awscli/customizations/dynamodb/subcommands.py", line 135, in _select
    self._dump_yaml(operation, response, parsed_globals)
  File "/Users/kyleknap/GitHub/aws-cli/awscli/customizations/dynamodb/subcommands.py", line 62, in _dump_yaml
    self._deserialize(operation_name, data)
  File "/Users/kyleknap/GitHub/aws-cli/awscli/customizations/dynamodb/subcommands.py", line 45, in _deserialize
    'AttributeValue'
  File "/Users/kyleknap/GitHub/aws-cli/awscli/customizations/dynamodb/transform.py", line 34, in transform
    model, params, transformation, target_shape)
  File "/Users/kyleknap/GitHub/aws-cli/awscli/customizations/dynamodb/transform.py", line 41, in _transform_parameters
    model, params, transformation, target_shape)
  File "/Users/kyleknap/GitHub/aws-cli/awscli/customizations/dynamodb/transform.py", line 56, in _transform_structure
    target_shape)
  File "/Users/kyleknap/GitHub/aws-cli/awscli/customizations/dynamodb/transform.py", line 41, in _transform_parameters
    model, params, transformation, target_shape)
  File "/Users/kyleknap/GitHub/aws-cli/awscli/customizations/dynamodb/transform.py", line 80, in _transform_list
    member_model, params[i], transformation, target_shape)
  File "/Users/kyleknap/GitHub/aws-cli/awscli/customizations/dynamodb/transform.py", line 41, in _transform_parameters
    model, params, transformation, target_shape)
  File "/Users/kyleknap/GitHub/aws-cli/awscli/customizations/dynamodb/transform.py", line 65, in _transform_map
    params[key] = transformation(value)
  File "/Users/kyleknap/GitHub/aws-cli/awscli/customizations/dynamodb/types.py", line 260, in deserialize
    return deserializer(value[dynamodb_type])
  File "/Users/kyleknap/GitHub/aws-cli/awscli/customizations/dynamodb/types.py", line 275, in _deserialize_b
    return Binary(value)
  File "/Users/kyleknap/GitHub/aws-cli/awscli/customizations/dynamodb/types.py", line 50, in __init__
    ', '.join([str(t) for t in BINARY_TYPES]))
TypeError: Value must be of the following types: <type 'bytearray'>, <type 'str'>.
2018-11-12 19:02:53,696 - MainThread - awscli.clidriver - DEBUG - Exiting with rc 255

Value must be of the following types: <type 'bytearray'>, <type 'str'>.
```
Any ideas why? I'm running on Python2.7

Reply:
Looks like this is because we're disabling blob parsing in scalarparse